drivers/sensor/max30208/max30208.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2021, Silvano Cortesi
+ * Copyright (c) 2021, ETH Zürich
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max30208
+
+#include <sys/byteorder.h>
+#include <logging/log.h>
+
+#include "max30208.h"
+
+LOG_MODULE_REGISTER(MAX30208, CONFIG_SENSOR_LOG_LEVEL);
+
+static int max30208_start_measurement(const struct device *dev)
+{
+	const struct max30208_config *config = dev->config;
+
+	/* Trigger one measurement */
+	int ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_TEMP_SETUP,
+					MAX30208_CONVERT_T_MASK);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int max30208_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+	struct max30208_data *data = dev->data;
+	uint8_t local_buffer[MAX30208_BYTES_PER_VAL] = { 0 };
+
+	/* If new value in ring buffer available, get it */
+	if (ring_buf_size_get(&data->raw_buffer) < MAX30208_BYTES_PER_VAL) {
+		return -ENODATA;
+	}
+
+	int ret = ring_buf_get(&data->raw_buffer, local_buffer, MAX30208_BYTES_PER_VAL);
+
+	if (ret < 0) {
+		LOG_ERR("Couldn't get data from ringbuffer");
+	}
+
+	/* Concat two bytes to a int16 value */
+	int16_t temp_val = (int16_t)sys_get_be16(local_buffer);
+
+	/* Convert raw value into temperature */
+	val->val1 = temp_val / MAX30208_ONE_DEGREE;
+	val->val2 = (temp_val % MAX30208_ONE_DEGREE) * MAX30208_LSB_e6;
+
+	return 0;
+}
+
+#ifndef CONFIG_MAX30208_TRIGGER
+static int max30208_data_ready(const struct device *dev, uint8_t *ready)
+{
+	const struct max30208_config *config = dev->config;
+	struct max30208_status status;
```

Comment:
why using a struct to read a single byte?

Reply:
Its a bitfield struct. As it implies the same runtime cost as using a raw byte, I think its better here as specific fields can be accessed by a meaningful name.